docs/Workshop.md

```diff
@@ -0,0 +1,311 @@
+2. A desktop version of Docker, such as [Docker Desktop](https://www.docker.com/products/docker-desktop/) (which is free for open source projects) or [Rancher Desktop](https://rancherdesktop.io/) running the Docker daemon.
+3. [Git](https://git-scm.com/).
+4. For Windows users only: [Windows Subsystem for Linux](https://learn.microsoft.com/en-us/windows/wsl/install).
+5. [Node.js](https://nodejs.org/en).
+6. [Scala base installation](https://docs.scala-lang.org/getting-started/index.html).
+7. Java 11+ (likely installed with [SDKMAN!](https://sdkman.io/)).
+8. [Intellij IDEA with the Scala plugin](https://docs.scala-lang.org/getting-started/intellij-track/getting-started-with-scala-in-intellij.html), [VSCode with metals](https://scalameta.org/metals/docs/editors/vscode/), or any other [metals-compatible editor](https://scalameta.org/metals/docs/).
+
+## How to Launch Docker Desktop 
+
+### For macOS Users
+
+1. Open the Finder.
+2. Navigate to the "Applications" folder.
+3. Double-click on "Docker.app" to start Docker Desktop.
+
+   - *Note:* A whale 🐳 icon in the top status bar signifies Docker Desktop is running and can be accessed from a terminal.
+
+### For Windows Users
+
+1. Click on the Windows start menu.
+2. Begin typing "Docker Desktop".
+3. When Docker Desktop appears in the search results, click on it to launch.
+
+   - *Note:* A Docker icon in the Windows system tray signifies Docker Desktop is running and can be accessed from a terminal.
+
+### For Linux Users (Using Ubuntu as an Example)
+
+1. Click on the Ubuntu 'Show Applications' button located at the bottom of the sidebar, or press "Super+A".
+2. Begin typing "Docker Desktop".
+3. When Docker Desktop appears in the search results, click on it to launch.
+
+   - *Note:* A Docker icon in the status bar signifies Docker Desktop is running and can be accessed from a terminal.
+
+## How to Run the Project
+
+To run the project, follow the steps below:
+
+```bash
+# Launch sbt
+sbt
+
+# Inside the sbt shell, enter the following command. This will incrementally compile the project and execute the `dockerComposeUp` task whenever a source file changes.
+~scaladays-workshop-2023/dockerComposeUp
+```
+
+## Infrastructure
+
+### build.sbt
+
+#### Project Structure
+
+[![](https://mermaid.ink/img/pako:eNqVkctqwzAQRX9FTLc2JMoioEIhfmwKWbl0U3chrEktrIeRx01NyL9Xdg0hu2Z3dTiagTsXaLxCEPAVZN-yt6x2jB0-ahgaaaSS05CefeiG1vcp3_BdDZ-zkc0Ghm8MK8gjIKLcaHS0siKyxlvr3ev7cWXljVUrOrA0fWHZLebLhiUWy-wllnM02nUVTQbZhg0UfIfiiWOj9tvk75metaJW8P7n-d7fPujzB_3dP3xIwGKwUqtY-GX-H0tr0WINIkaFJzma2F7trlGVI_lqcg0ICiMmMPZKEhZaxlPZe1gqTT6AOEkz4PUXfjyaBQ?type=png)](https://mermaid.live/edit#pako:eNqVkctqwzAQRX9FTLc2JMoioEIhfmwKWbl0U3chrEktrIeRx01NyL9Xdg0hu2Z3dTiagTsXaLxCEPAVZN-yt6x2jB0-ahgaaaSS05CefeiG1vcp3_BdDZ-zkc0Ghm8MK8gjIKLcaHS0siKyxlvr3ev7cWXljVUrOrA0fWHZLebLhiUWy-wllnM02nUVTQbZhg0UfIfiiWOj9tvk75metaJW8P7n-d7fPujzB_3dP3xIwGKwUqtY-GX-H0tr0WINIkaFJzma2F7trlGVI_lqcg0ICiMmMPZKEhZaxlPZe1gqTT6AOEkz4PUXfjyaBQ)
```

Comment:
We could use https://github.com/dwijnand/sbt-project-graph to generate the graph and link here the generated image.

![projects-graph](https://github.com/xebia-functional/tictactoe/assets/720923/4f4c51a3-d2b5-465d-b837-0404d3d090e3)

Reply:
Even use some online viewer:
https://dreampuf.github.io/GraphvizOnline/#digraph%20%22projects-graph%22%20%7B%0A%20%20%20%20graph%5Brankdir%3D%22LR%22%5D%0A%20%20%20%20node%20%5B%0A%20%20%20%20%20%20%20%20shape%3D%22record%22%0A%20%20%20%20%5D%0A%20%20%20%20edge%20%5B%0A%20%20%20%20%20%20%20%20arrowtail%3D%22none%22%0A%20%20%20%20%5D%0A%20%20%20%20%22commonJS%22%5Blabel%3D%3CcommonJS%3E%5D%0A%20%20%20%20%22commonJVM%22%5Blabel%3D%3CcommonJVM%3E%5D%0A%20%20%20%20%22scaladays-workshop-2023%22%5Blabel%3D%3Cscaladays-workshop-2023%3E%5D%0A%20%20%20%20%22server%22%5Blabel%3D%3Cserver%3E%5D%0A%20%20%20%20%22tttClient%22%5Blabel%3D%3CtttClient%3E%5D%0A%20%20%20%20%22scaladays-workshop-2023%22%20-%3E%20%22server%22%20%5Bstyle%3Ddashed%5D%0A%20%20%20%20%22scaladays-workshop-2023%22%20-%3E%20%22server%22%20%5Bstyle%3Dsolid%5D%0A%20%20%20%20%22scaladays-workshop-2023%22%20-%3E%20%22tttClient%22%20%5Bstyle%3Ddashed%5D%0A%20%20%20%20%22scaladays-workshop-2023%22%20-%3E%20%22tttClient%22%20%5Bstyle%3Dsolid%5D%0A%20%20%20%20%22server%22%20-%3E%20%22commonJVM%22%20%5Bstyle%3Dsolid%5D%0A%20%20%20%20%22tttClient%22%20-%3E%20%22commonJS%22%20%5Bstyle%3Dsolid%5D%0A%7D

Reply:
This actually uses mermaid, like your second graph. I had ai draw them for me, because @rafa was doing the graphs. I expect they will be replaced by his stuff.